pkg/instancetype/builder/builder.go

```diff
@@ -0,0 +1,146 @@
+package builder
```

Comment:
This could live in `pkg/libinstancetype` instead, to be consistent with other builders we have in the codebase (`libvmi`, `libdv`)? Might make it easier to extract it in the future?